compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -756,41 +756,114 @@ object Parsers {
      *   4. there is at least one token between the braces
      *   5. the closing brace is also at the end of the line, or it is followed by one of
      *      `then`, `else`, `do`, `catch`, `finally`, `yield`, or `match`.
-     *   6. the opening brace does not follow a `=>`. The reason for this condition is that
-     *      rewriting back to braces does not work after `=>` (since in most cases braces are omitted
-     *      after a `=>` it would be annoying if braces were inserted).
-     */
-    def bracesToIndented[T](body: => T, rewriteWithColon: Boolean): T = {
+     *   6. the opening brace does not follow a closing brace
```

Comment:
These conditions seem to be not in line with the spec:

> A template body can alternatively consist of a colon followed by one or more indented statements. To this purpose we introduce a new `<colon>` token that reads as
the standard colon "`:`" but is generated instead of it where `<colon>`
is legal according to the context free syntax, but only if the previous token
is an alphanumeric identifier, a backticked identifier, or one of the tokens `this`, `super`, "`)`", and "`]`".

(and the same holds for function arguments). The important part is the last: an inserted `:` can only appear after an alphanumeric identifier, a backticked identifier, or one of the tokens `this`, `super`, "`)`", and "`]`. That's stronger than the condition that the previous token may not be `}`. And I think if we implement that precisely, we can get rid of the annoying `inStatSeq` arguments.